.github/scripts/download_pretrained.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import time
+from typing import Optional, Text, Tuple
+
+from transformers import AutoTokenizer, TFAutoModel
+
+import rasa.shared.utils.io
+from rasa.nlu.utils.hugging_face.registry import model_weights_defaults
+
+COMP_NAME = "LanguageModelFeaturizer"
+DEFAULT_MODEL_NAME = "bert"
+
+
+def get_model_name_and_weights_from_config(
+    config_path: str,
+) -> Tuple[Optional[Text], Optional[Text]]:
+    config = rasa.shared.utils.io.read_config_file(config_path)
+    print(config)
+    steps = config.get("pipeline", [])
+
+    # Look for LanguageModelFeaturizer
+    steps = list(filter(lambda x: x["name"] == COMP_NAME, steps))
+
+    if len(steps) == 0:
+        print(f"No {COMP_NAME} found")
+        return None, None
+    elif len(steps) > 1:
+        print(f"Too many ({len(steps)}) {COMP_NAME}s found")
+        return None, None
+
+    lmfeat_step = steps[0]
+
+    if "model_name" not in lmfeat_step:
+        return DEFAULT_MODEL_NAME, model_weights_defaults[DEFAULT_MODEL_NAME]
+    model_name = lmfeat_step["model_name"]
+
+    model_weights = lmfeat_step.get("model_weights", model_weights_defaults[model_name])
+
+    return model_name, model_weights
+
+
+def instantiate_to_download(model_weights: Text) -> None:
```

Comment:
Depending on the `model_weights`, this function could throw Exceptions. Should those be caught so that the script can exit gracefully? The two failures I can think of are

1. The model doesn't exist, e.g. `model_weights = "abc"`, then `AutoTokenizer.from_pretrained` will raise a `requests.exceptions.HTTPError`.
2. The model exists, but has not Tensorflow weights, then `TFAutoModel.from_pretrained` will raise an `OSError`.

Reply:
Good catch 💪

I adjusted the code to be closer to https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py#L119-L123.
Also adjusted/added test cases.

Some errors, like `OSError` are not handled, because I think the error msg + stack trace leads the developer/user to the mistake.

Reply:
👍  Ah yes, very good point. Seeing those errors in the logs actually can help fixing wrong configs much easier.